tests/integration/synapseclient/models/async/test_file_async.py

```diff
@@ -82,6 +85,82 @@ async def test_store_in_project(self, project_model: Project, file: File) -> Non
         assert file.file_handle.key is not None
         assert file.file_handle.external_url is None
 
+    async def test_store_in_project_with_read_error(
+        self,
+        project_model: Project,
+        file: File,
+        mocker: MockerFixture,
+    ) -> None:
+        # GIVEN a file
+        file.name = str(uuid.uuid4())
+
+        # AND spy/mocks to simulate a read error
+        spy_tls_stream = mocker.spy(httpcore._backends.anyio.AnyIOStream, "start_tls")
+        logger_warning_spy = mocker.spy(self.syn.logger, "warning")
+        original_read = httpcore._backends.anyio.AnyIOStream.read
+        call_count = 0
+
+        async def new_read(*args, **kwargs) -> bytes:
+            nonlocal call_count
+            async_network_stream: httpcore.AsyncNetworkStream = (
+                spy_tls_stream.spy_return
+            )
+            call_count = call_count + 1
+            # This is very brittle, however, I do not see data to determine what the
```

Comment:
👀 Does that mean if we add functions called in the async stream that we would have edit this call_count?

Reply:
Yeah, if more or less calls occur during the `store_async` function we would need to adjust this call count.

This specific way of replacing the read isn't working on GH actions. Though I did confirm the logic is working on my local machine. This needs to be fixed, but more digging into this is needed. I can get the hostname/IP addresses used, but being this "far" down in the libraries and working directly in httpcore, I lose a lot of context like request headers and the uri.